politeiawww/dcc.go

```diff
@@ -538,3 +595,297 @@ func (p *politeiawww) processGetDCCs(gds cms.GetDCCs) (*cms.GetDCCsReply, error)
 		DCCs: dccs,
 	}, nil
 }
+
+func (p *politeiawww) processSupportOpposeDCC(sd cms.SupportOpposeDCC, u *user.User) (*cms.SupportOpposeDCCReply, error) {
```

Comment:
One last nit.  Can we move the validation that doesn't require any db calls (support string validation and signature validation) to the top?  It's better to handle the cheap validation first so that if they fail you can exit right away and not do those db calls unnecessarily.

Reply:
Sure thing

Reply:
Looks like you've still got the request validation below the getDCC db call.  It should look something like this:

```
func (p *politeiawww) processSupportOpposeDCC(sd cms.SupportOpposeDCC, u *user.User) (*cms.SupportOpposeDCCReply, error) {
	log.Tracef("processSupportOpposeDCC: %v %v", sd.Token, u.ID)

	// The submitted Vote in the request must either be "aye" or "nay"
	if sd.Vote != supportString && sd.Vote != opposeString {
		return nil, www.UserError{
			ErrorCode:    cms.ErrorStatusInvalidSupportOppose,
			ErrorContext: []string{"support string not aye or nay"},
		}
	}

	// Validate signature
	msg := fmt.Sprintf("%v%v", sd.Token, sd.Vote)
	err := validateSignature(sd.PublicKey, sd.Signature, msg)
	if err != nil {
		return nil, err
	}
```

Then the rest of the validation that requires db calls.

Reply:
Oh was overlooking the getDCC call.  Sorry.

Reply:
Still need to move the signature validation to the top.